bin/check-asg-instances-created.rb

```diff
@@ -57,41 +57,57 @@ class CheckAsgInstanceCreated < Sensu::Plugin::Check::CLI
          long:        '--critical_limit Critical Limit',
          description: 'Critical Limit for launching and terminated instances'
 
+  option :time_limit,
+         short:       '-t T',
+         long:        '--time Timelimit',
+         description: 'Time Limit for launching and terminated instances (default 1800s)'
+
   def asg
     @asg = Aws::AutoScaling::Client.new
   end
 
+  def ec2
+    @ec2 = Aws::EC2::Client.new
+  end
+
   def describe_activities(asg_group_name)
     asg.describe_scaling_activities(
       auto_scaling_group_name: asg_group_name.to_s
     )
   end
 
+  def describe_instance(instance_id)
+    ec2.describe_instances(
+      instance_ids: [instance_id]
+    )
+  end
+
   def run
-    warning = 3
-    critical = 4
-    result_launched = ''
+    warning = 2
+    critical = 3
+    time = 1800
     result_terminated = ''
-    instance_launching = 0
     instance_terminating = 0
     time_now = Time.now
     time_utc_offset = time_now - time_now.utc_offset
 
-    if !config[:warning_limit].nil?
+    unless config[:warning_limit].nil?
       warning = config[:warning_limit]
-    elsif !config[:critical_limit].nil?
+    end
+    unless config[:critical_limit].nil?
```

Comment:
If you want to go this route this is better: `critical = config[:critical_limit]  unless config[:critical_limit].nil?`